tests/test_spark_ui_endpoint.py

```
@@ -0,0 +1,188 @@
+    _stub_session(monkeypatch, "sparkpilot.aws_clients.session_for_environment", fake_client)
+
+    resp = client.get(f"/v1/runs/{run['id']}/spark-ui", follow_redirects=False)
+    assert resp.status_code == 302, resp.text
+    assert resp.headers["location"] == minted
+
+
+def test_eks_run_returns_404_no_ui_for_engine() -> None:
+    # EMR on EKS has no first-class presigned per-run UI: the client returns None regardless of
+    # dry-run, so this case needs no live client and no dry-run toggle.
+    client = TestClient(app)
+    _, op, _, run = _create_ready_environment_and_run(client, suffix="sparkui-eks")
```

Ruff (0.15.15)

[warning] Unpacked variable `op` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)